pytensor/tensor/subtensor.py

```diff
@@ -197,18 +230,37 @@ def get_canonical_form_slice(
     in which 0 <= start <= stop <= length and step > 0, and a flag which says
     if the resulting set of numbers needs to be reversed or not.
 
+    Returns
+    -------
+    slc
+        Slice or scalar variable.
+    direction
+        Direction to iterate the resulting elements in. (-1 or 1).
     """
     from pytensor.tensor import ge, lt, sign, switch
 
     if not isinstance(theslice, slice):
-        try:
-            value = as_index_literal(theslice)
-        except NotScalarConstantError:
-            value = theslice
-
-        value = switch(lt(value, 0), (value + length), value)
+        # Try to extract a scalar slice and return it already.
+        sslice: int | np.integer | ScalarVariable | None = None
+        if isinstance(theslice, ScalarVariable):
+            sslice = theslice
+        else:
+            try:
+                vlit = as_index_literal(theslice)
+                if vlit is None:
+                    return slice(0, length, 1), 1
+                if isinstance(vlit, slice):
+                    # Input was a SliceType variable, from which a slice was extracted.
+                    theslice = vlit
+                else:
+                    sslice = vlit
+            except NotScalarConstantError:
+                raise ValueError(f"Slice {theslice} is not a supported slice type.")
```

Comment:
Couldn't it be a slice with symbolic start/step/stop at this point?

Reply:
Okay this function does not expect symbolic slices (or at least it was never testing them), so the new changes shoudn't either?

If it's not a slice it must be a scalar index, and you shouldn't have to worry about the slice case showing up again here. If it should handle symbolic slices (i.e, SliceVariable) types, then we also need tests for those.

Reply:
It should however handle symbolic ScalarVariables (and possibly Tensor with ndim=0)

Reply:
That still means this function only needs to handle python slices, and the inner code can be simplified